Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp

```diff
@@ -466,11 +467,14 @@ static bool shouldAllowExternalLoad(const URL& url)
 
 static void* openFunc(const char* uri)
 {
-    ASSERT(XMLDocumentParserScope::currentCachedResourceLoader);
+    ASSERT(XMLDocumentParserScope::currentCachedResourceLoader());
     ASSERT(libxmlLoaderThread == &Thread::current());
 
-    CachedResourceLoader& cachedResourceLoader = *XMLDocumentParserScope::currentCachedResourceLoader;
-    Document* document = cachedResourceLoader.document();
+    RefPtr cachedResourceLoader = XMLDocumentParserScope::currentCachedResourceLoader().get();
+    if (!cachedResourceLoader)
+        return &globalDescriptor;
+
+    Document* document = cachedResourceLoader->document();
```

Comment:
Why aren't we using a RefPtr for the document here?

Reply:
Because the patch was to fix `CachedResourceLoader`.

The scope already expected quite a bit just doing that.  If I added `Document`, the patch would have gotten even larger.

Reply:
I'll make this change for this function.

Reply:
I'm also fine with you sticking to CachedResourceLoader in this patch.

Reply:
Since I was already modifying this line, I might as well make the change.  :)